component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java

```diff
@@ -134,18 +146,52 @@ public AuthenticatorFlowStatus process(HttpServletRequest request,
-            publishPostEmailOTPGeneratedEvent(request, context);
-            if (context.getProperty(EmailOTPAuthenticatorConstants.AUTHENTICATION)
-                    .equals(EmailOTPAuthenticatorConstants.AUTHENTICATOR_NAME)) {
-                // if the request comes with authentication is EmailOTP, it will go through this flow.
-                // set the current authenticator name
-                context.setCurrentAuthenticator(getName());
-                return AuthenticatorFlowStatus.INCOMPLETE;
+            User user;
+            AuthenticatedUser authenticatedUser = getAuthenticatedUser(context);
+            if (authenticatedUser == null) {
+                if (StringUtils.isEmpty(request.getParameter(EmailOTPAuthenticatorConstants.USER_NAME))) {
+                    redirectUserToIDF(response, context);
+                    context.setProperty(IS_IDF_INITIATED_FROM_AUTHENTICATOR, true);
+                    return AuthenticatorFlowStatus.INCOMPLETE;
+                } else {
+                    authenticatedUser = resolveUser(request, context);
+                    setResolvedUserInContext(context, authenticatedUser);
+                    user = getUser(authenticatedUser);
+                    if (user == null) {
+                        authenticatedUser = null;
+                    }
+                }
+            }
+            if (authenticatedUser != null && isPreviousIdPAuthenticationFlowHandler(context)) {
+                user = getUser(authenticatedUser);
+                if (user != null) {
+                    authenticatedUser = new AuthenticatedUser(user);
+                    setResolvedUserInContext(context, authenticatedUser);
+                } else {
+                    authenticatedUser = null;
+                }
+            }
+
+            if (authenticatedUser != null) {
+                initiateAuthenticationRequest(request, response, context);
+                publishPostEmailOTPGeneratedEvent(request, context);
+                if (context.getProperty(EmailOTPAuthenticatorConstants.AUTHENTICATION)
+                        .equals(EmailOTPAuthenticatorConstants.AUTHENTICATOR_NAME)) {
+                    // if the request comes with authentication is EmailOTP, it will go through this flow.
+                    // set the current authenticator name
+                    context.setCurrentAuthenticator(getName());
+                    return AuthenticatorFlowStatus.INCOMPLETE;
+                } else {
+                    // if the request comes with authentication is basic, complete the flow.
+                    return AuthenticatorFlowStatus.SUCCESS_COMPLETED;
+                }
             } else {
-                // if the request comes with authentication is basic, complete the flow.
-                return AuthenticatorFlowStatus.SUCCESS_COMPLETED;
+                if (log.isDebugEnabled()) {
+                    log.debug("The user does not exist in the user stores.");
+                }
```

Comment:
We don't need to check if debug is enabled if we do not perform any expensive operations like string concatenation for the logging string.